src/main/java/gregtech/common/metatileentities/multi/electric/generator/LargeCombustionEngineWorkableHandler.java

```diff
@@ -71,4 +76,35 @@ public void deserializeNBT(NBTTagCompound compound) {
         super.deserializeNBT(compound);
         this.currentCycle = compound.getInteger("Cycle");
     }
+
+    @Override
+    protected boolean isObstructed() {
+        return checkIntakesObstructed();
+    }
+
+    private boolean checkIntakesObstructed() {
+        if(this.metaTileEntity instanceof MultiblockWithDisplayBase) {
+            MultiblockWithDisplayBase controller = (MultiblockWithDisplayBase) this.metaTileEntity;
+
+            EnumFacing facing = controller.getFrontFacing();
+            //TODO, this also needs to check Y for freedom wrench implementation
```

Comment:
This can mostly likely be checked with methods in RelativeDirection.

Reply:
The `Y` direction can be checked, but I was more talking about the `XY` and `ZY` permutations